diff --git a/api/molecules-container.hh b/api/molecules-container.hh
index 294a839e6..41ffc7e87 100644
--- a/api/molecules-container.hh
+++ b/api/molecules-container.hh
@@ -944,7 +944,7 @@ public:
    //!         success flag.)
    //!
    //! @return a list of atom names and their associated computed COD atom types
-   //! (a `cod::atom_type_t` for each atom, which holds nb1nb2/main_type/cod_type types,
+   //! (a `cod::atom_type_t` for each atom, which holds nb1nb2/main_type/full_type types,
    //! the nb2-extra-electrons and the hash value), return an empty list on failure
 #ifdef MAKE_ENHANCED_LIGAND_TOOLS
    std::vector<std::pair<std::string, cod::atom_type_t> > get_computed_acedrg_atom_types(const std::string &compound_id, int imol_enc);
diff --git a/coot-utils/cod-db-access.cc b/coot-utils/cod-db-access.cc
index ed965c549..8e7e64172 100644
--- a/coot-utils/cod-db-access.cc
+++ b/coot-utils/cod-db-access.cc
@@ -46,7 +46,7 @@ public:
 
 
 // fill data_store with as a std::vector<double>
-// 
+//
 int db_select_callback(void *data_store, int argc, char **argv, char **azColName) {
    int status = 0;
    std::vector<double> *v = static_cast<std::vector<double> * > (data_store);
@@ -150,7 +150,7 @@ get_atom_index(const cod::atom_type_t &cod_1, sqlite3 *db) {
 
    if (db) {
       std::string cmd = "SELECT atom_index from COD_TYPE_4_INDICES WHERE level_4_atom_type = " +
-         coot::util::single_quote(cod_1.cod_type, "'");
+         coot::util::single_quote(cod_1.full_type, "'");
       int index = -1; // unset
       void *data_pointer = static_cast<void *>(&index);
       char *zErrMsg = 0;
@@ -158,12 +158,12 @@ get_atom_index(const cod::atom_type_t &cod_1, sqlite3 *db) {
       if (rc !=  SQLITE_OK) {
          if (zErrMsg) {
             std::cout << "ERROR: processing command " << cmd << " " << zErrMsg << std::endl;
-         } else { 
+         } else {
             std::cout << "ERROR: processing command " << cmd << std::endl;
             sqlite3_free(zErrMsg);
          }
       } else {
-         std::cout << "    " << cod_1.cod_type << " " << index << std::endl;
+         std::cout << "    " << cod_1.full_type << " " << index << std::endl;
          idx = index;
       }
    }
@@ -284,7 +284,7 @@ get_bond_record_try_type_2(const cod::atom_type_t &cod_1,
       if (rc_1 !=  SQLITE_OK) {
          if (zErrMsg_1) {
             std::cout << "ERROR: processing command " << cmd_1 << " " << zErrMsg_1 << std::endl;
-         } else { 
+         } else {
             std::cout << "ERROR: processing command " << cmd_1 << std::endl;
             sqlite3_free(zErrMsg_1);
          }
@@ -344,7 +344,7 @@ get_type_3_bonds_inner(const std::vector<int> &cod_1_type_3_atom_indices,
       if (one_or_two_is_vector == 1)
          cmd = "SELECT mean, sd, count from COD_TYPE_4_BONDS WHERE atom_index_1 = " +
             coot::util::int_to_string(cod_1_type_3_atom_indices[i]) + " AND atom_index_2 = " +
-            coot::util::int_to_string(idx_2) + 
+            coot::util::int_to_string(idx_2) +
             " AND hybridization = " + coot::util::single_quote(hybridization_info.string, "'") +
             " AND ring = " + coot::util::single_quote(ring_bool_str, "'");
       if (one_or_two_is_vector == 2)
@@ -480,8 +480,8 @@ get_bond_record_try_type_3(const cod::atom_type_t &cod_1,
    cod::bond_table_record_t btr;
    //
    bool done_2 = false;
-   if (cod_1.cod_type == cod_1.main_type) {
-      if (cod_2.cod_type == cod_2.main_type) {
+   if (cod_1.full_type == cod_1.main_type) {
+      if (cod_2.full_type == cod_2.main_type) {
          btr = get_bond_record_try_type_2(cod_1, cod_2, idx_1, idx_2, n_count_min, db);
          done_2 = true;
       }
@@ -510,7 +510,7 @@ get_bond_record_try_type_3(const cod::atom_type_t &cod_1,
       if (rc_1 !=  SQLITE_OK) {
          if (zErrMsg_1) {
             std::cout << "ERROR: processing command " << cmd_1 << " " << zErrMsg_1 << std::endl;
-         } else { 
+         } else {
             std::cout << "ERROR: processing command " << cmd_1 << std::endl;
             sqlite3_free(zErrMsg_1);
          }
@@ -689,7 +689,7 @@ get_hybridization_info_string(RDKit::RWMol &rdkm, unsigned int mol_atom_idx_1, u
       current++;
       n_bonds_at_2++;
    }
-   
+
    if (at_1->getAtomicNum() == 8) {
       if (at_2->getAtomicNum() == 15) {
          if (n_bonds_at_1 == 1)
@@ -742,7 +742,7 @@ get_ring_info_bool_str(RDKit::RingInfo* ring_info_p, int idx_1, int idx_2) {
 void debug_atom_types(const RDKit::RWMol &rdkm,
                       const std::vector<cod::atom_type_t> &v,
                       const coot::dictionary_residue_restraints_t &dict) {
-   
+
    if (v.size() != dict.number_of_atoms()) {
       std::cout << "debug_atom_types(): mismatch types " << v.size() << " "
                 << dict.number_of_atoms() << std::endl;
@@ -753,7 +753,7 @@ void debug_atom_types(const RDKit::RWMol &rdkm,
          at->getProp("name", name);
          std::string ee_type = v[i].nb1nb2.extra_electron_type();
          std::cout << " types for " << name
-                   << " type-4 " << v[i].cod_type
+                  << " type-4 " << v[i].full_type
                    << " nee " << v[i].nb1nb2.n_extra_electrons()
                    << " std-level_2: " << std::setw(20) << v[i].nb1nb2.string()
                    << " ee: " << ee_type
@@ -824,8 +824,8 @@ void validate_bonds(mmdb::Residue *residue_p,
                   // cod::bond_table_record_t btr = get_bond_table_record(cod_1, cod_2, db);
 
                   std::cout << "##### " << atom_name_1 << " " << atom_name_2 << " types:  l4: "
-                            << cod_1.cod_type << "  l2: " << cod_1.nb1nb2.extra_electron_type() << " and l4: "
-                            << cod_2.cod_type << "  l2: " << cod_2.nb1nb2.extra_electron_type() << std::endl;
+                            << cod_1.full_type << "  l2: " << cod_1.nb1nb2.extra_electron_type() << " and l4: "
+                            << cod_2.full_type << "  l2: " << cod_2.nb1nb2.extra_electron_type() << std::endl;
 
                   hybridization_info_t hybridization_info =
                      get_hybridization_info_string(rdkm, atom_1.first, atom_2.first);
@@ -865,12 +865,12 @@ void validate_bonds(mmdb::Residue *residue_p,
 
                         }
                      } else {
-                        std::cout << "No db atom type index for atom type " << cod_2.cod_type << std::endl;
+                        std::cout << "No db atom type index for atom type " << cod_2.full_type << std::endl;
                         // something like:
                         // cod::bond_table_record_t btr = get_bond_record_try_type_3(cod_1, cod_2, -1, -1, n_count_min, db);
                      }
                   } else {
-                     std::cout << "No db atom type index for atom type " << cod_1.cod_type << std::endl;
+                     std::cout << "No db atom type index for atom type " << cod_1.full_type << std::endl;
                      // so... what do we do now?
                   }
 
@@ -952,7 +952,7 @@ int main(int argc, char **argv) {
                   // (otherwise the atom types are not correct).
                   coot::dictionary_residue_restraints_t dict(residue_p);
 
-                  validate_bonds_for_residue_dict_db_file_name(residue_p, dict, 
+                  validate_bonds_for_residue_dict_db_file_name(residue_p, dict,
                                                                reprotonate_flag, db_file_name);
 
                } else {
@@ -985,7 +985,7 @@ int main(int argc, char **argv) {
                      geom.get_monomer_restraints(rmit.monomer_idx);
                   int imol_enc = coot::protein_geometry::IMOL_ENC_ANY;
                   mmdb::Residue *residue_p = dict.GetResidue(false, 20);
-                  validate_bonds_for_residue_dict_db_file_name(residue_p, dict, 
+                  validate_bonds_for_residue_dict_db_file_name(residue_p, dict,
                                                                reprotonate_flag, db_file_name);
                } else {
                   std::cout << "WARNING:: bad monomer index " << std::endl;
diff --git a/coot-utils/dictionary-atom-types.cc b/coot-utils/dictionary-atom-types.cc
index 004779488..1fc1176a1 100644
--- a/coot-utils/dictionary-atom-types.cc
+++ b/coot-utils/dictionary-atom-types.cc
@@ -1,18 +1,18 @@
 /* lidia-core/dictionary-atom-types.cc
- * 
+ *
  * Copyright 2016 by Medical Research Council
  * Author: Paul Emsley
- * 
+ *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
  * the Free Software Foundation; either version 3 of the License, or (at
  * your option) any later version.
- * 
+ *
  * This program is distributed in the hope that it will be useful, but
  * WITHOUT ANY WARRANTY; without even the implied warranty of
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
  * General Public License for more details.
- * 
+ *
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
@@ -21,7 +21,7 @@
 
 #ifndef MAKE_ENHANCED_LIGAND_TOOLS
 int main(int argc, char **argv) {return 0;}
-#else 
+#else
 #include "lidia-core/cod-atom-types.hh"
 
 #include <map>
@@ -37,7 +37,7 @@ bool string_int_pair_sorter(const std::pair<std::string, unsigned int> &p1,
                             const std::pair<std::string, unsigned int> &p2) {
 
    return (p2.second < p1.second);
-} 
+}
 
 int find_string_in_vector(const std::vector<std::pair<std::string, unsigned int> > &v,
                           const std::string &t) {
@@ -46,10 +46,10 @@ int find_string_in_vector(const std::vector<std::pair<std::string, unsigned int>
    for (unsigned int i=0; i<v.size(); i++) {
       if (v[i].first == t) {
          return i;
-      } 
+      }
    }
    return r;
-} 
+}
 
 
 int main(int argc, char **argv) {
@@ -89,16 +89,16 @@ int main(int argc, char **argv) {
       if (! mol) {
          std::cout << "Null mol from mol_from_dictionary() for " <<  comp_id << std::endl;
       } else {
-         
+
          mmdb::Residue *residue_p = coot::util::get_first_residue(mol);
 
          if (! residue_p) {
             // pretty strange
             std::cout << "Null residue from mol from mol_from_dictionary() for "
                       << comp_id << std::endl;
-         } else { 
+         } else {
 
-            try { 
+            try {
                RDKit::RWMol rdkm = coot::rdkit_mol_sanitized(residue_p, imol, geom);
 
                cod::atom_types_t t;
@@ -107,20 +107,20 @@ int main(int argc, char **argv) {
                   // std::cout << comp_id << " was good" << std::endl;
                   for (unsigned int iat=0; iat<r.atom_info.size(); iat++) {
                      const std::string &key = r.atom_info[iat].type_energy;
-                     if (0) 
+                     if (0)
                         std::cout << comp_id << "  "
                                   << std::setw(4) << r.atom_info[iat].atom_id << "  "
-                                  << std::setw(6) << r.atom_info[iat].type_energy << "   " 
-                                  << v[iat].cod_type << "\n";
+                                  << std::setw(6) << r.atom_info[iat].type_energy << "   "
+                                  << v[iat].full_type << "\n";
 
                      it = atom_map.find(key);
                      if (it == atom_map.end()) {
-                        std::pair<std::string, unsigned int> p(v[iat].cod_type, 1);
+                        std::pair<std::string, unsigned int> p(v[iat].full_type, 1);
                         atom_map[key].push_back(p);
-                     } else { 
-                        int idx = find_string_in_vector(atom_map[key], v[iat].cod_type);
+                     } else {
+                        int idx = find_string_in_vector(atom_map[key], v[iat].full_type);
                         if (idx == -1) { // not found
-                           std::pair<std::string, unsigned int> p(v[iat].cod_type, 1);
+                           std::pair<std::string, unsigned int> p(v[iat].full_type, 1);
                            atom_map[key].push_back(p);
                         } else {
                            atom_map[key][idx].second++;
@@ -134,21 +134,21 @@ int main(int argc, char **argv) {
             }
             catch (const std::runtime_error &rte) {
                std::cout << "error:: rte " << rte.what() << " " << comp_id << std::endl;
-            } 
+            }
             catch (const std::exception &e) {
                std::cout << "error:: exception " << e.what() << " " << comp_id << std::endl;
-            } 
+            }
          }
       }
    }
 
    bool output_atom_map = false;
-   if (output_atom_map) { 
+   if (output_atom_map) {
       std::map<std::string, std::vector<std::pair<std::string, unsigned int> > >::iterator itv;
       for (itv=atom_map.begin(); itv!=atom_map.end(); itv++) {
          std::sort(itv->second.begin(), itv->second.end(), string_int_pair_sorter);
          std::cout << itv->first << "       ";
-         for (unsigned int i=0; i<itv->second.size(); i++) { 
+         for (unsigned int i=0; i<itv->second.size(); i++) {
             std::cout << "   " << itv->second[i].first  << " " << itv->second[i].second;
          }
          std::cout << "\n";
@@ -165,7 +165,7 @@ int main(int argc, char **argv) {
    // vector of refmac energy types (of which should only be one, of
    // course)
 
-   if (0) { 
+   if (0) {
       std::map<std::string, std::vector<std::pair<std::string, unsigned int> > >::iterator itv;
       std::map<std::string, std::vector<std::string> > cod_map;
       std::map<std::string, std::vector<std::string> >::const_iterator it_cod;
@@ -178,7 +178,7 @@ int main(int argc, char **argv) {
       for (it_cod=cod_map.begin(); it_cod!=cod_map.end(); it_cod++) {
          if (it_cod->second.size() != 1) {
             std::cout << "strange cod: " << std::left << std::setw(20) << it_cod->first << "     ";
-            for (unsigned int i=0; i<it_cod->second.size(); i++) { 
+            for (unsigned int i=0; i<it_cod->second.size(); i++) {
                std::cout << std::setw(5) << it_cod->second[i] << " ";
             }
             std::cout << "\n";
@@ -205,23 +205,23 @@ int main(int argc, char **argv) {
       if (! mol) {
          std::cout << "Null mol from mol_from_dictionary() for " <<  comp_id << std::endl;
       } else {
-         
+
          mmdb::Residue *residue_p = coot::util::get_first_residue(mol);
 
          if (! residue_p) {
             // pretty strange
             std::cout << "Null residue from mol from mol_from_dictionary() for "
                       << comp_id << std::endl;
-         } else { 
+         } else {
 
-            try { 
+            try {
                RDKit::RWMol rdkm = coot::rdkit_mol_sanitized(residue_p, imol, geom);
                cod::atom_types_t t;
                std::vector<cod::atom_type_t> v = t.get_cod_atom_types(rdkm);
                if (v.size() == r.atom_info.size()) {
                   for (unsigned int iat=0; iat<r.atom_info.size(); iat++) {
                      const std::string &te = r.atom_info[iat].type_energy;
-                     const std::string &key = v[iat].cod_type;
+                     const std::string &key = v[iat].full_type;
 
                      it = reverse_atom_map.find(key);
 
@@ -243,28 +243,28 @@ int main(int argc, char **argv) {
             }
             catch (const std::runtime_error &rte) {
                std::cout << "error:: rte " << rte.what() << " " << comp_id << std::endl;
-            } 
+            }
             catch (const std::exception &e) {
                std::cout << "error:: exception " << e.what() << " " << comp_id << std::endl;
-            } 
+            }
          }
       }
    }
 
    bool output_reverse_atom_map = true;
-   if (output_reverse_atom_map) { 
+   if (output_reverse_atom_map) {
       std::map<std::string, std::vector<std::pair<std::string, unsigned int> > >::iterator itv;
       for (itv=reverse_atom_map.begin(); itv!=reverse_atom_map.end(); itv++) {
          std::sort(itv->second.begin(), itv->second.end(), string_int_pair_sorter);
          std::cout << "cod type " << std::left << std::setw(20) << itv->first
                    << "       energy-lib-types: ";
-         for (unsigned int i=0; i<itv->second.size(); i++) { 
+         for (unsigned int i=0; i<itv->second.size(); i++) {
             std::cout << "   " << itv->second[i].first  << " " << itv->second[i].second;
          }
          std::cout << "\n";
       }
    }
-   
+
    return r;
 }
 
diff --git a/coot-utils/test-cod-atom-types.cc b/coot-utils/test-cod-atom-types.cc
index a191335ee..fbb7923c3 100644
--- a/coot-utils/test-cod-atom-types.cc
+++ b/coot-utils/test-cod-atom-types.cc
@@ -1,18 +1,18 @@
 /* lidia-core/test-cod-atom-types.cc
- * 
+ *
  * Copyright 2016 by Medical Research Council
  * Author: Paul Emsley
- * 
+ *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
  * the Free Software Foundation; either version 3 of the License, or (at
  * your option) any later version.
- * 
+ *
  * This program is distributed in the hope that it will be useful, but
  * WITHOUT ANY WARRANTY; without even the implied warranty of
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
  * General Public License for more details.
- * 
+ *
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
@@ -86,7 +86,7 @@ void write_types(RDKit::RWMol &rdkm) {
          //
          if (false)
             std::cout << " " << std::right << std::setw(3) << iat << "    "
-                      << atom_ele << "    " << name << "     " << v[iat].cod_type
+                      << atom_ele << "    " << name << "     " << v[iat].full_type
                       << "\n";
 
          // Now I want to check the tier-2 types too
@@ -94,7 +94,7 @@ void write_types(RDKit::RWMol &rdkm) {
          std::cout << " " << std::right << std::setw(3) << iat << "    "
                    << atom_ele << "    " << name << "     "
                    << v[iat].nb1nb2.string() << "  "
-                   << v[iat].cod_type << "  "
+                   << v[iat].full_type << "  "
                    << "\n";
       }
       catch (const KeyErrorException &err) { }
@@ -204,7 +204,7 @@ void molecule_from_SMILES(const std::string &smiles_string) {
 
          std::cout << "PE-TYPES:: -------- got " << v.size() << " atoms " << std::endl;
          for (unsigned int i=0; i<v.size(); i++)
-            std::cout << "   " << i << " " << v[i].cod_type << "" << std::endl;
+            std::cout << "   " << i << " " << v[i].full_type << "" << std::endl;
       }
    }
    catch (const std::runtime_error &rte) {
@@ -224,8 +224,8 @@ write_bonds_by_type(RDKit::RWMol *rdkm, const std::vector<cod::atom_type_t> &v)
             const RDKit::Bond *bond_p = rdkm->getBondWithIdx(ib);
             int idx_1 = bond_p->getBeginAtomIdx();
             int idx_2 = bond_p->getEndAtomIdx();
-            std::string t1 = v[idx_1].cod_type;
-            std::string t2 = v[idx_2].cod_type;
+            std::string t1 = v[idx_1].full_type;
+            std::string t2 = v[idx_2].full_type;
             if (t1 > t2) {
                // std::swap(t1, t2);
             }
@@ -248,7 +248,7 @@ void molecule_from_mdl_mol(const std::string &file_name) {
          std::cout << "PE-TYPES:: -------- " << v.size() << " atoms " << " from "
                    << file_name << std::endl;
          for (unsigned int i=0; i<v.size(); i++)
-            std::cout << "   " << i << " " << v[i].cod_type << "" << std::endl;
+            std::cout << "   " << i << " " << v[i].full_type << "" << std::endl;
          write_bonds_by_type(rdkm, v);
       } else {
          std::cout << "WARNING:: file " << file_name << " null molecule" << std::endl;
diff --git a/lidia-core/bond-record-container-t.cc b/lidia-core/bond-record-container-t.cc
index 63a6255d4..19339a868 100644
--- a/lidia-core/bond-record-container-t.cc
+++ b/lidia-core/bond-record-container-t.cc
@@ -1,18 +1,18 @@
 /* lidia-core/bond-record-container-t.cc
- * 
+ *
  * Copyright 2016 by Medical Research Council
  * Author: Paul Emsley
- * 
+ *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
  * the Free Software Foundation; either version 3 of the License, or (at
  * your option) any later version.
- * 
+ *
  * This program is distributed in the hope that it will be useful, but
  * WITHOUT ANY WARRANTY; without even the implied warranty of
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
  * General Public License for more details.
- * 
+ *
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
@@ -38,7 +38,7 @@
 // #include "coot-utils/coot-coord-utils.hh" out of order now
 
 // can throw std::runtime_error.
-// 
+//
 // The input atom names are directly from the pdbx cif file, so the
 // don't contain spaces, so the input at_name_in should not contain
 // spaces.
@@ -50,7 +50,7 @@ cod::bond_record_container_t::get_atom_index(const std::string &at_name_in,
 
    unsigned int idx = 0;
    bool found = false;
-   
+
    unsigned int n_mol_atoms = mol.getNumAtoms();
    for (unsigned int iat=0; iat<n_mol_atoms; iat++) {
       const RDKit::Atom *at_p = mol[iat];
@@ -73,7 +73,7 @@ cod::bond_record_container_t::get_atom_index(const std::string &at_name_in,
       std::string m(std::string("atom name \"") + at_name_in +
 		    std::string("\" not found in dictionary atom name list"));
       throw std::runtime_error(m);
-   } 
+   }
    return idx;
 }
 
@@ -86,7 +86,7 @@ cod::bond_record_container_t::get_atom_index(const std::string &at_name_1,
 
    unsigned idx = 0;
    bool found = false;
-   for (unsigned int iat=0; iat<rest.atom_info.size(); iat++) { 
+   for (unsigned int iat=0; iat<rest.atom_info.size(); iat++) {
       if (rest.atom_info[iat].atom_id_4c == at_name_1) {
 	 found = true;
 	 break;
@@ -268,7 +268,7 @@ cod::bond_record_container_t::read_acedrg_table_dir(const std::string &dir_name)
    for (unsigned int i=0; i<tables.size(); i++) {
       const std::string &table = tables[i];
       std::string fn = coot::util::file_name_non_directory(table);
-      if (fn != "bond_idx.table") { 
+      if (fn != "bond_idx.table") {
 	 bond_record_container_t single;
 	 bool success = single.read_acedrg_table(table);
 	 if (success)
@@ -382,7 +382,7 @@ cod::bond_record_container_t::read(const std::string &atom_type_indices_file_nam
    if (types.size())
       success = read_bonds(bonds_file_name, types);
    return success;
-} 
+}
 
 
 std::vector<std::string>
@@ -393,7 +393,7 @@ cod::bond_record_container_t::read_atom_type_indices(const std::string &atom_typ
    std::ifstream f(atom_type_indices_file_name.c_str());
    if (f) {
       std::string line;
-      try { 
+      try {
 	 while (std::getline(f, line)) {
 	    int idx = coot::util::string_to_int(line.substr(0, 6));
 	    types[idx] = line.substr(7);
@@ -420,13 +420,13 @@ cod::bond_record_container_t::read_bonds(const std::string &bonds_file_name,
    // running getline() on the file takes 0.2s,
    // 1.8 s is spend on string -> number
    // the atom types look up takes ~0.1s each (really?)
-   
+
    bool status = true;
    std::ifstream f(bonds_file_name.c_str());
    if (f) {
       // bonds.reserve(563711); // nice speedup.
       std::string line;
-      try { 
+      try {
 	 while (std::getline(f, line)) {
 
 	    if (true) {
@@ -452,7 +452,7 @@ cod::bond_record_container_t::read_bonds(const std::string &bonds_file_name,
 
 	       if (type_2 < type_1)
 		  std::swap(type_1, type_2);
-	       
+
 	       bond_table_record_t bond(type_1, type_2, bl, std_dev, count);
 
 	       std::string l3_type_1 = type_1.main_type;
@@ -468,13 +468,13 @@ cod::bond_record_container_t::read_bonds(const std::string &bonds_file_name,
 		   << " - failed to parse: " << line << " " << rte.what() << std::endl;
       }
 
-      if (false) { 
+      if (false) {
 	 int n_forward = 0;
 	 int n_backward = 0;
 	 int n_equal = 0;
 	 int n_many  = 0;
 
-	 for (unsigned int i=0; i<bonds.size(); i++) { 
+	 for (unsigned int i=0; i<bonds.size(); i++) {
 	    if (bonds[i].cod_type_2 < bonds[i].cod_type_1)
 	       n_forward++;
 	    if (bonds[i].cod_type_1 < bonds[i].cod_type_2)
@@ -490,11 +490,11 @@ cod::bond_record_container_t::read_bonds(const std::string &bonds_file_name,
 	 std::cout << "           n_equal    " << n_equal    << std::endl;
 	 std::cout << "           n_many     " << n_many     << std::endl;
       }
-      
+
       if (false)
 	 for (unsigned int i=0; i<20; i++)
 	    std::cout << "   " << bonds[i] << std::endl;
-	 
+
 
    } else {
       status = false;
@@ -507,10 +507,10 @@ void
 cod::bond_record_container_t::check() const {
 
    std::cout << "start check " << std::endl;
-   
+
    std::map<std::string, std::map<std::string, std::map<std::string, std::map<std::string, std::vector<bond_table_record_t> > > > >::const_iterator it_l2_a;
    std::map<std::string, std::map<std::string, std::map<std::string, std::vector<bond_table_record_t> > > >::const_iterator it_l2_b;
-   
+
    for (it_l2_a=bonds_map.begin(); it_l2_a!=bonds_map.end(); it_l2_a++) {
       std::string key_l2_a = it_l2_a->first;
 
@@ -544,12 +544,12 @@ cod::bond_record_container_t::get_is_hydrogen_flags(const RDKit::RWMol &rdkm) co
 }
 
 // atom types are generated from the atoms of rest (which contains hydrogens)
-// 
+//
 void
 cod::bond_record_container_t::validate(mmdb::Residue *res,
 				       const coot::dictionary_residue_restraints_t &rest) const {
 
-   if (res) { 
+   if (res) {
       std::string res_name = res->GetResName();
       std::cout << "validate: " << res_name << " " << rest.residue_info.comp_id
 		<< std::endl;
@@ -557,7 +557,7 @@ cod::bond_record_container_t::validate(mmdb::Residue *res,
       if (res_name == rest.residue_info.comp_id) {
 
 	 try {
-	    
+
 	    RDKit::RWMol rdkm = coot::rdkit_mol(rest);
 	    coot::rdkit_mol_sanitize(rdkm);
 
@@ -607,7 +607,7 @@ cod::bond_record_container_t::validate(mmdb::Residue *res,
 		  std::string at_name_1 = coot::util::remove_whitespace(bond.atom_id_1());
 		  std::string at_name_2 = coot::util::remove_whitespace(bond.atom_id_2());
 
-		  if (false) { 
+		  if (false) {
 		     std::cout << "in validate(): at_name_1 is \"" << at_name_1
 			       << "\"" << std::endl;
 		     std::cout << "               at_name_2 is \"" << at_name_2
@@ -615,9 +615,9 @@ cod::bond_record_container_t::validate(mmdb::Residue *res,
 		  }
 
 		  try {
-		     
+
 		     // use atom indices and look in v. Use non-whitespace names
-		     // 
+		     //
 		     unsigned int atom_idx_1 = get_atom_index(at_name_1, rdkm);
 		     unsigned int atom_idx_2 = get_atom_index(at_name_2, rdkm);
 
@@ -636,13 +636,13 @@ cod::bond_record_container_t::validate(mmdb::Residue *res,
 
 			if (cod_type_2.full_type < cod_type_1.full_type)
 			   std::swap(cod_type_1, cod_type_2);
-		     
+
 			bond_table_record_t cod_bond =
 			   get_cod_bond_from_table(cod_type_1, cod_type_2);
-		     
+
 			double dist =
 			   get_bond_distance_from_model(at_name_1_4c, at_name_2_4c, res);
-			
+
 			double z = 9999;
 			if (cod_bond.std_dev > 0)
 			   z = std::abs((cod_bond.mean-dist)/cod_bond.std_dev);
@@ -659,8 +659,8 @@ cod::bond_record_container_t::validate(mmdb::Residue *res,
 				  << std::setw(8) << z
 				  << std::endl;
 
-			   // << cod_type_1.main_type << "  " << cod_type_1.cod_type << " "
-			   // << cod_type_2.main_type << "  " << cod_type_2.cod_type
+			   // << cod_type_1.main_type << "  " << cod_type_1.full_type << " "
+			   // << cod_type_2.main_type << "  " << cod_type_2.full_type
 		     }
 		  }
 		  catch (const std::runtime_error &rte) {
@@ -672,7 +672,7 @@ cod::bond_record_container_t::validate(mmdb::Residue *res,
 	    } else {
 	       std::cout << "mismatch between number of atoms in molecule and "
 			 << "COD types list" << std::endl;
-	    } 
+	    }
 	 }
 	 catch(const std::runtime_error &rte) {
 	    std::cout << "error::" << rte.what() << std::endl;
@@ -681,10 +681,10 @@ cod::bond_record_container_t::validate(mmdb::Residue *res,
 	 std::cout << "Mismatch in residue name vs dictionary comp_id "
 		   << res_name << " " << rest.residue_info.comp_id
 		   << std::endl;
-      } 
+      }
    } else {
       std::cout << "Null res" << std::endl;
-   } 
+   }
 
 }
 
@@ -704,14 +704,14 @@ cod::bond_record_container_t::get_cod_bond_from_table(const cod::atom_type_t &co
 		<< cod_type_2.nb1nb2.string() << "   "
 		<< cod_type_2.main_type << std::endl;
    }
-      
+
    std::map<std::string, std::map<std::string, std::map<std::string, std::map<std::string, std::vector<bond_table_record_t> > > > >::const_iterator it_l2_a;
    std::map<std::string, std::map<std::string, std::map<std::string, std::vector<bond_table_record_t> > > >::const_iterator it_l2_b;
-   
+
    std::map<std::string, std::map<std::string, std::vector<bond_table_record_t> > >::const_iterator it_l3_a;
 
    std::map<std::string, std::vector<bond_table_record_t> >::const_iterator it_l3_b;
-   
+
    it_l2_a = bonds_map.find(cod_type_1.nb1nb2.string());
    if (it_l2_a != bonds_map.end()) {
       it_l2_b = it_l2_a->second.find(cod_type_2.nb1nb2.string());
@@ -736,7 +736,7 @@ cod::bond_record_container_t::get_cod_bond_from_table(const cod::atom_type_t &co
 	       bond_table_record_t::approximation_level_t approx_level = bond_table_record_t::EXTRA_ELECTRON;
 	       bond = make_bond_from_level_2_map(cod_type_1, cod_type_2, it_l2_b->second, approx_level);
 	       found_bond = true;
-	       
+
 	       // std::string m("missing cod_type_2 level_3 " + cod_type_1.nb1nb2.string());
 	       // throw(std::runtime_error(m));
 	    }
@@ -823,9 +823,9 @@ cod::bond_record_container_t::make_bond_from_level_3_vector(const cod::atom_type
 
 	 // local bonds: collect a set of bond records that have exact
 	 // match for cod_type_1
-	 // 
+	 //
 	 std::vector<bond_table_record_t> local_bonds;
-	 for (unsigned int i=0; i<v.size(); i++) { 
+	 for (unsigned int i=0; i<v.size(); i++) {
 	    if (cod_type_1 == v[i].cod_type_1) {
 	       // do we need a "distance" metric here before we add this?
 	       local_bonds.push_back(v[i]);
@@ -844,7 +844,7 @@ cod::bond_record_container_t::make_bond_from_level_3_vector(const cod::atom_type
 
 	 // likewise but the other testing the other atom
 	 std::vector<bond_table_record_t> local_bonds;
-	 for (unsigned int i=0; i<v.size(); i++) { 
+	 for (unsigned int i=0; i<v.size(); i++) {
 	    if (cod_type_2 == v[i].cod_type_2) {
 	       local_bonds.push_back(v[i]);
 	    }
@@ -926,7 +926,7 @@ cod::bond_record_container_t::t3_miss_diagnose(const atom_type_t &cod_type_1,
 
    std::map<std::string, std::map<std::string, std::map<std::string, std::map<std::string, std::vector<bond_table_record_t> > > > >::const_iterator it_l2_a;
    std::map<std::string, std::map<std::string, std::map<std::string, std::vector<bond_table_record_t> > > >::const_iterator it_l2_b;
-   
+
    std::map<std::string, std::map<std::string, std::vector<bond_table_record_t> > >::const_iterator it_l3_a;
 
    std::map<std::string, std::vector<bond_table_record_t> >::const_iterator it_l3_b;
@@ -935,7 +935,7 @@ cod::bond_record_container_t::t3_miss_diagnose(const atom_type_t &cod_type_1,
    bool found_bond = false;
 
    bool vector_found = false;
-   
+
    for (it_l2_a=bonds_map.begin(); it_l2_a!=bonds_map.end(); it_l2_a++) {
       for (it_l2_b=it_l2_a->second.begin(); it_l2_b!=it_l2_a->second.end(); it_l2_b++) {
 	 for (it_l3_a=it_l2_b->second.begin(); it_l3_a != it_l2_b->second.end(); it_l3_a++) {
@@ -997,7 +997,7 @@ cod::bond_record_container_t::t3_miss_diagnose(const atom_type_t &cod_type_1,
       std::cout << "::::::: Hmmmmm t3_miss_diagnose() bond not found "
 		<< std::endl;
    }
-   
+
    if (vector_found) {
       std::cout << "::::::: Hmmmmm t3_miss_diagnose() found vector "
 		<< std::endl;
@@ -1005,9 +1005,9 @@ cod::bond_record_container_t::t3_miss_diagnose(const atom_type_t &cod_type_1,
       std::cout << "::::::: Hmmmmm t3_miss_diagnose() vector not found "
 		<< std::endl;
    }
-   
+
 }
-      
+
 
 
 
@@ -1152,7 +1152,7 @@ cod::bond_record_container_t::db_add_level_4_types(sqlite3 *db) {
    unsigned int n = cod_atom_type_map.size();
 
    for (unsigned int i=0; i<n; i++) {
-      std::string level_4 = it_map->first.cod_type;
+		std::string level_4 = it_map->first.full_type;
       std::string level_3 = it_map->first.main_type;
       std::string level_2 = it_map->first.nb1nb2.string();
       std::string degree_colon_type = it_map->first.nb2_extra_els_str();
@@ -1273,15 +1273,15 @@ cod::bond_record_container_t::db_add_level_4_types(sqlite3 *db) {
 		  cmd += ", ";
 		  cmd += coot::util::int_to_string(btr.count);
 		  cmd += ");";
-      
+
 		  char *zErrMsg = 0;
 		  int rc = sqlite3_exec(db, cmd.c_str(), db_callback, 0, &zErrMsg);
 		  if (rc !=  SQLITE_OK) {
 		     if (zErrMsg) {
 			std::cout << "ERROR: processing command: " << cmd << " " << zErrMsg
 				  << " for "
-				  << btr.cod_type_1.cod_type << " "
-				  << btr.cod_type_2.cod_type
+				  << btr.cod_type_1.full_type << " "
+				  << btr.cod_type_2.full_type
 				  << btr.hybridization_token << " "
 				  << btr.ring_token << " from "
 				  << btr.file_name << " line " << btr.line_number
