diff --git a/coot.desktop b/coot.desktop
index 10f1efac8..1ddcbc847 100644
--- a/coot.desktop
+++ b/coot.desktop
@@ -1,9 +1,12 @@
 [Desktop Entry]
 Type=Application
 Name=Coot
+GenericName=Crystallographic Object-Oriented Toolkit
 Comment=Macromolecular model builder
 Exec=coot %f
-Icon=coot
-Terminal=true
-Categories=Science;Chemistry;
+Icon=io.github.pemsley.coot
+StartupWMClass=coot-1
+Comment=Crystallographic Object-Oriented Toolkit
+Terminal=false
+Categories=Science;Chemistry;Biology;GTK;
 MimeType=chemical/x-pdb;chemical/x-mmcif;chemical/x-mdl-molfile;
